### PR TITLE
Redis prober

### PR DESCRIPTION
**Pull request by wejdross**
## Summary

* Adding Redis prober to sliexporter

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [x] Link this PR to related issues.

### CODE REVIEW

**Review by zugao — CHANGES_REQUESTED** [review]
I tried the prober and it successfully exports redis metrics.

File: pkg/sliexporter/vshnpostgresql_controller/vshnpostgresql_controller.go [hunk 1]
```diff
@@ -55,6 +55,9 @@ type probeManager interface {
 //+kubebuilder:rbac:groups=vshn.appcat.vshn.io,resources=vshnpostgresqls,verbs=get;list;watch
 //+kubebuilder:rbac:groups=vshn.appcat.vshn.io,resources=vshnpostgresqls/status,verbs=get
 
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch
```
[the remark this answers is not in the document]
Comment (wejdross) [reply pkg/sliexporter/vshnpostgresql_controller/vshnpostgresql_controller.go:58]: this is important for VSHNPostgreSQL as it ensures correct RBAC, it's duplicated across Redis and PostgreSQL, but generator is quite clever and understands that correctly

File: pkg/sliexporter/vshnredis_controller/vshnredis_controller.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	certPair, err := tls.X509KeyPair(credSecret.Data["tls.crt"], credSecret.Data["tls.key"])
+	if err != nil {
+		return nil, err
+	}
+	tlsConfig := tls.Config{
+		Certificates: []tls.Certificate{certPair},
+		RootCAs:      x509.NewCertPool(),
+		// this line must be set to true otherwise probe fails with:
+		// "failed to verify certificate: x509: certificate is valid for vshn.appcat.vshn.ch, not redis-headless.instance-name.svc.cluster.local"
+		InsecureSkipVerify: true,
```
[the remark this answers is not in the document]
Comment (wejdross) [reply pkg/sliexporter/vshnredis_controller/vshnredis_controller.go:126]: If You try to connect without TLS it fails, if You configure TLS and set InsecureSkipVerify: false it fails with message from comment above, if You configure TLS, skip verification (different names), then it works, I just made it work, but I think we should take a look at root cause in separate story

[the remark this answers is not in the document]
Comment (Kidswiss) [reply pkg/sliexporter/vshnredis_controller/vshnredis_controller.go:126]: @zugao I agree, the prober should connect via valid TLS if the connection requires it. The connection secret should contain all the information necessary to do that.

Why do you use `x509.NewCertPool()` on line 123? The credential secret should give you the ca as well under the `ca.crt` field. Maybe that's the root of your issue?

[the remark this answers is not in the document]
Comment (wejdross) [reply pkg/sliexporter/vshnredis_controller/vshnredis_controller.go]: fixed using this commit: https://github.com/vshn/component-appcat/pull/204

after that I was able to remove insecureSkipVerify flag as dns names matched

File: cmd/sliexporter.go [hunk 3]
```diff
@@ -39,6 +41,11 @@ func init() {
 		"Enabling this will ensure there is only one active controller manager.")
 	SLIProberCMD.Flags().BoolVar(&s.enableVSHNPostgreSQL, "vshn-postgresql", getEnvBool("APPCAT_SLI_VSHNPOSTGRESQL"),
 		"Enable probing of VSHNPostgreSQL instances")
+	// SLIProberCMD.Flags().BoolVar(&s.enableVSHNRedis, "vshn-redis", getEnvBool("APPCAT_SLI_VSHNREDIS"),
```
[the remark this answers is not in the document]
Comment (wejdross) [reply cmd/sliexporter.go]: done :)

**Review by zugao — APPROVED** [review]
LGTM

**Pull request merged**